tests/functional/detection_status/test_detection_status.py

```diff
@@ -0,0 +1,148 @@
+                          'hello',
+                          self._test_dir)
+
+    def test_same_file_change(self):
+        """
+        This tests the change of the detection status of bugs when the file
+        content changes.
+        """
+
+        # Check the first file version
+        self._create_source_file(1)
+        reports = self._cc_client.getRunResults(1, 100, 0, {}, {})
+        self.assertEqual(len(reports), 2)
+        self.assertTrue(all(map(
+            lambda r: r.detectionStatus == shared.ttypes.DetectionStatus.NEW,
+            reports)))
+
+        # Check the second file version
+        self._create_source_file(2)
+        reports = self._cc_client.getRunResults(1, 100, 0, {}, {})
+        for report in reports:
+            if report.bugHash in ['209be2f6905590d99853ce01d52a78e0',
+                                  'e8f47588c8095f02a53e338984ce52ba']:
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.UNRESOLVED)
+            elif report.bugHash == 'cbd629ba2ee25c41cdbf5e2e336b1b1c':
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.NEW)
+
+        # Check the third file version
+        self._create_source_file(3)
+        reports = self._cc_client.getRunResults(1, 100, 0, {}, {})
+        for report in reports:
+            if report.bugHash == 'cbd629ba2ee25c41cdbf5e2e336b1b1c':
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.UNRESOLVED)
+            elif report.bugHash == 'ac147b31a745d91be093bd70bbc5567c':
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.NEW)
+            elif report.bugHash in ['209be2f6905590d99853ce01d52a78e0',
+                                    'e8f47588c8095f02a53e338984ce52ba']:
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.RESOLVED)
+
+        # Check the third file version
+        self._create_source_file(2)
+        reports = self._cc_client.getRunResults(1, 100, 0, {}, {})
+        for report in reports:
+            if report.bugHash == 'cbd629ba2ee25c41cdbf5e2e336b1b1c':
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.UNRESOLVED)
+            elif report.bugHash in ['209be2f6905590d99853ce01d52a78e0',
+                                    'e8f47588c8095f02a53e338984ce52ba']:
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.REOPENED)
+            elif report.bugHash == 'ac147b31a745d91be093bd70bbc5567c':
+                self.assertEqual(report.detectionStatus,
+                                 shared.ttypes.DetectionStatus.RESOLVED)
```

Comment:
please also verify the source file content referred to by the bug.
for the resolved bug (i.e. i=1/0 we should see version 2) we should see the last version where the bug occurred, for new, unresolved, and reopened, we should always see the latest file content.